### PR TITLE
add debug

### PR DESCRIPTION
**Pull request by yevgenfs**
add debug to the project

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -50,3 +50,31 @@ modules.order
 Module.symvers
 Mkfile.old
 dkms.conf
+*.js
+*.html
+*.png
+*.css
+build/CMakeFiles/stm32_spinner_project.dir/link.txt
+build/Debug/stm32_spinner_project.bin
+build/.cmake/api/v1/reply/codemodel-v2-266db49fb6621a6af4fd.json
```
Comment (oleh-plotnikov) [inline .gitignore]: you  can specify build/* instead these files

File: .vscode/launch.json [hunk 2]
```diff
@@ -6,7 +6,7 @@
             "type": "cortex-debug",
             "request": "launch",
             "cwd": "${workspaceRoot}/build",
-            "executable": "C:\\Users\\yevhen.surkov\\Documents\\STM_WORKPLACE\\stm32_spinner_project-2\\build\\stm32_spinner_project-2.elf",
+            "executable": "C:\\Users\\yevhen.surkov\\Documents\\STM_WORKPLACE\\stm32_spinner_project\\build\\Debug\\stm32_spinner_project.elf",
```
Comment (oleh-plotnikov) [inline .vscode/launch.json]: Use ${workspaceRoot}/

File: CMakeLists.txt [hunk 3]
```diff
@@ -1,12 +1,205 @@
...
+  ${HAL_PATH}/Src/stm32f4xx_hal_spi.c
+  ${HAL_PATH}/Inc/stm32f4xx_hal_spi.h
+  ${HAL_PATH}/Src/stm32f4xx_hal_rcc.c
+  ${HAL_PATH}/Inc/stm32f4xx_hal_rcc.h
+  ${HAL_PATH}/Src/stm32f4xx_hal_flash_ex.c
+  ${HAL_PATH}/Inc/stm32f4xx_hal_flash_ex.h
+  ${HAL_PATH}/Src/stm32f4xx_hal_rcc_ex.c
+  ${HAL_PATH}/Inc/stm32f4xx_hal_rcc_ex.h
+  ${HAL_PATH}/Src/stm32f4xx_hal_i2c.c
+  ${HAL_PATH}/Inc/stm32f4xx_hal_i2c.h
```
Comment (oleh-plotnikov) [inline CMakeLists.txt]: You don't need to add include files. just use 
set(INCLUDE_DIRS ${INCLUDE_DIRS} ${....)
or/and
add_subdirectory(...)

**Review by oleh-plotnikov — CHANGES_REQUESTED** [review]

File: .gitignore (hunk 1 above)
Comment (yevgenfs) [reply .gitignore]: done

File: .vscode/launch.json (hunk 2 above)
Comment (yevgenfs) [reply .vscode/launch.json]: done

File: CMakeLists.txt (hunk 3 above)
Comment (yevgenfs) [reply CMakeLists.txt]: done

**Pull request merged**